lib/Reader/readerir.cpp

```diff
@@ -5994,8 +5994,36 @@ void GenIR::switchOpcode(IRNode *Opr) {
   BasicBlock *CurrBlock = LLVMBuilder->GetInsertBlock();
   TerminatorInst *TermInst = CurrBlock->getTerminator();
   SwitchInst *SwitchInstruction = cast<SwitchInst>(TermInst);
-
   SwitchInstruction->setCondition(Opr);
+
+  llvm::Type* OprType = Opr->getType();
+  LLVMContext &LLVMContext = *this->JitContext->LLVMContext;
+  Type *Int32Ty = Type::getInt32Ty(LLVMContext);
+  if (OprType != Int32Ty) {
+    // When we made the case values we assumed int32 as the type for the 
+    // values. But now we have an operand of a different type. LLVM 
+    // requires the values and the switch value to have the same type.
+    // So change the type of the values to be the type of the selector.
+    assert(OprType->isIntegerTy() && "Switch selector is not an integer type");
+    llvm::IntegerType* OprIntType = (llvm::IntegerType*)OprType;
+
+    // If a small selector type was used, e.g. i8, the type might not
+    // be big enough to hold all the values. But the values for the
+    // selector is popped off the evaluation stack, so it should be at
+    // least 32 bits. So check that.
+    unsigned OprBitWidth = OprIntType->getBitWidth();
+    assert((OprBitWidth > 32) && "Selector bit width is less then 32");
```

Comment:
We already tested that OprType is not Int32Ty, so I don't think the = 32 case would apply.